localstack/services/edge.py

```diff
@@ -97,19 +92,16 @@ def forward_request(self, method, path, data, headers):
             re.sub(r"^([^:]+://[^/]+).*", r"\1", orig_req_url) or "http://%s" % host
         )
 
-        # extract API details
-        api, port, path, host = get_api_from_headers(headers, method=method, path=path, data=data)
+        # re-create an HTTP request from the given parts
+        request = create_request_from_parts(method, path, data, headers)
+        from localstack.aws.protocol.service_router import determine_aws_service_name
```

Comment:
nit: I'm assuming this is here to avoid a circular import issue? If so, maybe we can reverse the order and make this a top-level import, and lazily load the imports in `service_router.py` instead. Feels like `localstack.aws.protocol.*` is the more general and `localstack.services.edge` the more specific module (hence the latter should be able to depend on the former). (not critical though, also totally fine to leave as-is!)

Reply:
Unfortunately, I overlooked this comment. Since we're already in the release process, I'll tackle this in the next iteration.